pycognaize/common/field_collection.py

```diff
@@ -20,4 +21,20 @@ def groups(self) -> dict:
     def groups_by_name(self, name: str) -> dict:
         """Returns fields that are contained in group with the
         given group name"""
-        return self.groups[name]
+        if name in self.groups.keys():
+            return self.groups[name]
+        else:
+            raise KeyError
+
+    def groups_by_field(self, field: Field) -> dict:
+        """Returns groups that contain the given field"""
+        try:
+            field_group_name = field.group_name
+            return self.groups_by_name(field_group_name)
+        except AttributeError or KeyError:
```

Comment:
`except (AttributeError, KeyError)`

Reply:
On the other hand, it is much better to avoid the exception altogether. Check the availability of the attribute beforehand. Same for the KeyError.